File: ios/Folio/Domain/Models/Tag.swift
import Foundation
import SwiftData

@Model
final class Tag {
    @Attribute(.unique) var id: UUID
    @Attribute(.unique) var name: String
    var isAIGenerated: Bool
    var articleCount: Int
    var articles: [Article]
    var createdAt: Date
    var serverID: String?

    init(name: String, isAIGenerated: Bool = false) {
        self.id = UUID()
        self.name = name
        self.isAIGenerated = isAIGenerated
        self.articleCount = 0
        self.articles = []
        self.createdAt = Date()
        self.serverID = nil
    }
}

File: server/internal/domain/tag.go
package domain

import "time"

type Tag struct {
	ID            string    `json:"id"`
	Name          string    `json:"name"`
	UserID        *string   `json:"user_id,omitempty"`
	IsAIGenerated bool      `json:"is_ai_generated"`
	ArticleCount  int       `json:"article_count"`
	CreatedAt     time.Time `json:"created_at"`
}
